dali/pipeline/operators/reader/loader/sequence_loader.h

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATORS_READER_LOADER_SEQUENCE_LOADER_H_
+#define DALI_PIPELINE_OPERATORS_READER_LOADER_SEQUENCE_LOADER_H_
+
+#include <numeric>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "dali/common.h"
+#include "dali/pipeline/operators/reader/loader/loader.h"
+
+namespace dali {
+
+struct TensorSequence {
+  std::vector<Tensor<CPUBackend>> tensors;
+};
+
+// TODO(klecki) consider using FileLoader as base class
+// TODO(klecki) ?? allow for more high level grouping of sequences:
+//              we should probably make sure that other Loaders read
+//              sequentially, and allow for SequenceLoader to wrap any other
+//              loader, similar for parser and reader
+// TODO(klecki) loader is responsible for handling shuffle_
+class SequenceLoader : public Loader<CPUBackend, TensorSequence> {
```

Comment:
Still, this class lacks some unit tests, e.g. for `CalculateStreamSize` or `PrepareEmpty`...

Reply:
How do you see testing private methods of a class? Should I start adding tests as a friend? Or maybe, should I change the approach and start making everything public?

Also, should I really write a tests for a method that has 3 lines?

The basic behaviour is tested by SequenceReader Op, this is not really a public API.

Reply:
Until we have consistent way to approach tests and test infrastructure ready I think we can leave it as it is.

Reply:
I'll create a Jira tasks for it, so it won't be forgotten